File: Crud2new/Crud2new/app/src/main/java/com/example/crud2new/Model/Usuario.kt
package com.example.crud2new.Model

data class Usuario(

    val nombre: String = "",
    val apellidos: String = "",
    val correo: String = "",
    val contraseña: String = "",
    val camisetas: List<Camiseta>

){
    val _id: String = ""

    constructor( camisetas: List<Camiseta>) : this("", "", "", "", camisetas)

}

File: TFG_API/modelo/usuario.ts
import mongoose, { Schema, Document } from 'mongoose';
import IUser from '../interfaces/usuario';

// Define el esquema de usuario
const SchemaUsuario: Schema = new Schema<IUser>(
  {
    nombre: { type: Schema.Types.String, required: true },
    apellidos: { type: Schema.Types.String, required: true },
    correo: { type: Schema.Types.String, required: true },
    contraseña: { type: Schema.Types.String, required: true },
    camisetas: {
      type: [Schema.Types.ObjectId],
      ref: "camisetas",
    },
  },
  { timestamps: false, versionKey: false }
);

export const UsuarioModel = mongoose.model<IUser & Document>(
  "usuarios",
  SchemaUsuario
);
